docs/proposals/excluding-drives.md

```diff
@@ -0,0 +1,108 @@
+# Proposal: Specific drives excluding
+
+Last updated: 07.02.2022
+
+
+## Abstract
+
+CSI Baremetal needs to have API for excluding specific drives from the common storage pool. 
+Excluded drives won't be used for allocating new Volumes or LVGs.
+
+## Background
+
+CSI customers might not want to use some disks for whatever reason:
+- for another application not managed by CSI
+- as a workaround for any disk detection issues in CSI
+
+So in these cases CSI haven't to allocate Volumes for some drives.
+
+## Proposal
+
+### User API
+
+The rules to include/exclude drives based on:
+- Size
+- Type (HDD, SSD, ...)
+- Serial Number
+- PID
+- VID
+```yaml
+drive_rule:
+  size:
+    less_then: 1Gi
+    more_then: 100Mi
+  # include drives with the following parameters only
+  in:
+    type:
+      - HDD
+    SN:
+      - sn1
+      - sn2
+    PID:
+      - ...
+    VID:
+      - ...
+  # exclude drives with the following parameters
+  not_in:
+    type:
+      - SSD
+    SN:
+      - ...
+    PID:
+      - ...
+    VID:
+      - ...
+```
+
+Settings can be set for all nodes in the cluster or for only one node. 
+In the node section user chooses is global rules be applied or not.
+```yaml
+drive_options:
+  global:
+    drive_rule: ...
+  nodes:
+    - name: node1
+      enable_global: true/false
+      drive_rule:
+    - name: ...
+```
+
+#### Pass settings
+Propose to process excluded drives on Discover stage in csi-baremetal-node part. 
+Node container will take drive setting from the ConfigMap created with csi-baremetal-deployment.
+
+Node ConfigMap can be updated in the following ways:
+1. Set global drive rule on CSI installation.
+
+Helm command for `csi-baremetal-driver` should be modified with:
+```yaml
+--set driver.node.global_drive_rule.size.more_then="1Gi" \
+--set driver.node.global_drive_rule.in.SN={sn1, sn2, sn3}
```

Comment:
Did I get it right that a node will have only one "in" or/and "not_in" list, and every "--set" command will change that list(s)?
From my prospective, it would be beneficial to have add/remove commands to modify settings from that lists.

Reply:
> Did I get it right that a node will have only one "in" or/and "not_in" list, and every "--set" command will change that list(s)?
No, we can use both options together with different parameters. For example
```
in:
  type: HDD
not_in:
  SN: {sn1, sn2}
```
So CSI will use all HDD drives except ones with sn1 or sn2 serial numbers. But yes, correct config will be user responsibility.

Reply:
> From my prospective, it would be beneficial to have add/remove commands to modify settings from that lists.

These options are for installation via helm only. It doesn't work well with a list. So these rules will be empty and we just have an option to set a full list. I think it will be helpful only in the case when we know cluster architecture in automation

Reply:
@safronovD , from previous experience, I'd strongly recommend to consider an ability to extend/modify lists/maps of drives which can be exposed/deprecated for CSI usage because we may have a huge panopticon of HW configuration once installation number has started growing. Can you please clarify, is that doable theoretically?

Reply:
In proposal 2 options are described:
- configure helm values on CSI installation
- editing setting ConfigMap to modify rules after installation

I guess your recomdadtion is cover by the second option

Reply:
It looks we are done here.

Reply:
The only think I just want to point out. It's better to have user friendly interface than making a user to edit configs with possibility of creation incorrect configuration.